src/rag/embedding.py

```
@@ -0,0 +1,80 @@
+            result = self.client.models.embed_content(
+                model=self.model_name,
+                contents=batch, # type: ignore
+                config=types.EmbedContentConfig(
+                    task_type=task_type.value,
+                    output_dimensionality=self.dimension,
+                ),
+            )
+                
+            if not result.embeddings:
+                self.logger.error("Batch returned no embeddings. Filling with zeros.")
+                raise ValueError('This embedding dimension is not 3072, there is something wrong about embedding')
```

Ruff (0.14.14)

[warning] Abstract `raise` to an inner function

(TRY301)